tests/pytest/test_defrag.py

```diff
@@ -0,0 +1,83 @@
+import time
+import json
+from RLTest import Defaults
+
+Defaults.decode_responses = True
+
+def enableDefrag(env):
+    # make defrag as aggressive as possible
+    env.cmd('CONFIG', 'SET', 'hz', '100')
+    env.cmd('CONFIG', 'SET', 'active-defrag-ignore-bytes', '1')
+    env.cmd('CONFIG', 'SET', 'active-defrag-threshold-lower', '0')
+    env.cmd('CONFIG', 'SET', 'active-defrag-cycle-min', '99')
+
+    try:
+        env.cmd('CONFIG', 'SET', 'activedefrag', 'yes')
```

Comment:
@MeirShpilraien i understand that unlike the redis defrag tests, this one isn't actually creating fragmentation and waiting for it to go down, which is ok.
and i understand that it does test all sorts of various types of json objects, and watches for stats about callback cycles, but how does it make sure that the pointers were really moved? what if redis forgets to call the module api callback?

maybe in addition to these, add one test that create holes and waits for them to get filled?

Reply:
Modify to tests to actually cause fragmentation and verify that defrag actually took place.

Reply:
if i read it correctly, you created a lot of jason fields, and then deleted all of them.
what i think you should do:
1. disable active defrag
2. create a lot of jason fields
3. delete all the even ones (or alternatively when you create them, you can create two keys, and interchangeably add fields to both, and then delete one key)
4. verify you managed to create fragmentation beyond a certain threshold.
5. enable active defrag.
6. wait for fragmentation levels to go down

see memefficiency.tcl in redis.
what i wanna see is that fields get actually re-located (not just that the callback is called)